externs/browser/url.js

```diff
@@ -25,7 +25,7 @@
 /**
  * @constructor
  * @implements {Iterable<!Array<string>>}
- * @param {(string|!URLSearchParams)=} init
+ * @param {(string|!URLSearchParams|Object<*, *>)=} init
```

Comment:
Hmmm, I'm looking at the spec here.
https://url.spec.whatwg.org/#interface-urlsearchparams
And the MDN docs here
https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams

The `init` argument is very flexible, type-wise, so I think something like this is needed.

```javascript
/**
 * @typedef {IArrayLike<string>}
 */
var URLSearchParamsTupleType;

/**
 * @typedef {IArrayLike<!UrlSearchParamsTupleType>|Iterable<!URLSearchParamsTupleType>}
 */
var URLSearchParamsTupleSequenceType;

/**
 * @constructor
 * @implements {Iterable<!Array<string>>}
 * @param {(string|!URLSearchParamsTupleSequenceType|!Object<string,string>)=} init
 */
function URLSearchParams(init) {}
```
Since `URLSearchParams` implements `Iterable<!Array<string>>` it is a subtype of the `URLSearchParamsTupleSequenceType` I defined above. We don't need to explicitly name it as an option.

BTW, I notice that MDN says `URLSearchParams` is "experimental" and the compatibility table at the bottom of the doc page says it definitely doesn't work in IE and has unknown status in Safari.

Reply:
The table seems to be a bit outdated/missing some information (it also uses the old compatibility data format, which probably means it was not touched in a while). It works in all of the latest versions of all of the modern browsers - I checked using BrowserStack.
(Note - Internet Explorer 11, released in 2013, is not modern ;))

Reply:
I'm willing to back off of the more elaborate code I suggested above.

What use-cases are you actually wanting to enable with this PR?
Could you provide sample code showing what they would look like.
Then we can work from there to get annotations that work.
I find the spec for this method and the MDN docs rather obtuse,
so I think concrete examples would be very helpful.

Reply:
The use case is very simple,
```
new URLSearchParams({parameter1: 1, parameter2: 2})
```


Some routing code I have -
```
 async goTo(path, queryParameters)
 {
  const canNavigate = await this._isNavigationAllowed();
  if (canNavigate === false)
  {
   return;
  }

  const url = new URL(path, this._locationProvider.href);
  if (queryParameters && queryParameters instanceof URLSearchParams)
  {
   url.search = queryParameters.toString();
  }
  else if (typeof queryParameters === kECMAScript.basicTypes.object)
  {
   url.search = new URLSearchParams(queryParameters).toString();
  }
  this._commitURL(null, null, url);
  this._handleLocationChange();
 }
```

And usage -
```
router.goTo(
 kApplication.routePaths.settings,
 {error: kApplication.errors.authorizationFailed});
```

Reply:
I realize that the use case does not justify `Object<*, *>`. I basically want numbers and others to be able to function as stringified keys here (in order to be able to also create a query string result of `1=8&2=9&true=k&toStringOfObject`).
Maybe this can be more explicit...
Object<(string|number|boolean), (string|number|boolean)>

On the other hand, if you choose Closure Compiler with advanced optimizations, you usually prefer to be type accurate... So maybe Object<string, string> is the right way to go.